nowcasting_api/database_fast.py

```diff
@@ -60,6 +62,9 @@ def get_forecast_values_all_compact(
     # join with model table
     query = query.filter(ForecastValueLatestSQL.model_id.in_(model_ids))
 
+    # make sure start time is set, and is not before the max history length
+    start_datetime_utc = get_start_datetime(start_datetime=start_datetime_utc)
+
```

Comment:
I think this should go in the [gsp.py](https://github.com/openclimatefix/uk-pv-national-gsp-api/blob/main/nowcasting_api/gsp.py#L104) function. 

I agree it should have a default, but perhaps in the gsp.py function

I was trying to make this function have pure database logic, not api logic. This fits in line with @devsjc belief and idea that datamodel function can be moved into datamodel library

Reply:
Perhaps [here](https://github.com/openclimatefix/uk-pv-national-gsp-api/blob/main/nowcasting_api/gsp.py#L110) we could say,`start_datetime_utc = get_start_datetime(start_datetime=start_datetime_utc)`

Reply:
Good point, better to keep this new DB function clean and concerns separate! Have amended 👍